pkg/kubectl/cmd/label.go

```diff
@@ -57,6 +57,9 @@ $ kubectl label pods foo bar-`
 )
 
 func NewCmdLabel(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	p := kubectl.NewHumanReadablePrinter(false, false, false, false, []string{})
```

Comment:
Not all resource types are defined statically in Kube code.  Our commands
use the provided Factory to abstract static references to Kube specific
resources, because we anticipate in the future our commands will *not*
reference types defined in Go code, but types defined dynamically by
retrieving swagger API descriptions from a server.  So having commands in
Kubectl that are generic (label, annotate, get, delete) specifically
reference kubectl.NewHumanReadablePrinter is incorrect.  If you need to get
the list of types that the printer supports, you should add a method to
'pkg/kubectl/cmd/util/factory.go' that returns the list of types supported
for a particular behavior.  If you are looking for all the types that the
kubectl command supports, you should probably add a method to RESTMapper
which is AllResources (that can return a list of objects with the resource
name, the kind, and a display name).  That is the default abstraction we
have in place to avoid coupling kubectl to statically defined Go types.

On Mon, Aug 24, 2015 at 5:58 AM, Janet Kuo <notifications@github.com> wrote:

> In pkg/kubectl/cmd/label.go
> <https://github.com/kubernetes/kubernetes/pull/13047#discussion_r37731172>
> :
>
> > @@ -57,6 +57,9 @@ $ kubectl label pods foo bar-`
> >  )
> >
> >  func NewCmdLabel(f *cmdutil.Factory, out io.Writer) *cobra.Command {
> > +	p := kubectl.NewHumanReadablePrinter(false, false, false, false, []string{})
>
> @smarterclayton <https://github.com/smarterclayton> I'm not sure if I
> understand it... NewHumanReadablePrinter is used here to retrieve all
> resource types (pod, service, etc.) that we handle, which should be the
> same for kubectl no matter which printer is used. Can you please clarify
> that for me?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13047/files#r37731172>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift